pkg/kubelet/kubelet.go

```diff
@@ -795,6 +802,9 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 			Name:   kl.nodeName,
 			Labels: map[string]string{"kubernetes.io/hostname": kl.hostname},
 		},
+		Spec: api.NodeSpec{
```

Comment:
@mikedanese Having it be part of Spec makes it worse, because then it's trivially easy to flip the value and run pods on the master.

@dchen1107 Folks running the OSS code may want to provide API access (but not direct machine access) to their cluster to other users (e.g. here's a copy of my kubeconfig but you don't have ssh access to the master). This change means that you can't effectively do that because I could run whatever I want on the master node.